bot-lib/src/commands/aur_search.rs

```diff
@@ -0,0 +1,88 @@
+use crate::data::PoiseContext;
+use color_eyre::eyre::Result;
+use itertools::Itertools;
+use poise::{serenity_prelude as serenity, CreateReply};
+use raur::Raur;
+use std::cmp::Reverse;
+
+const SEARCHCAP: usize = 20;
+const BASE_AUR_URL: &str = "https://aur.archlinux.org/packages/";
+#[poise::command(slash_command, prefix_command, rename = "aur")]
+pub async fn aur_search(ctx: PoiseContext<'_>, search: String, amount: usize) -> Result<()> {
+    ctx.defer().await?;
+
+    let search: String = search
+        .to_uppercase()
+        .chars()
+        .filter(|c| c.is_alphanumeric())
+        .collect::<String>();
+
+    if search.is_empty() {
+        ctx.reply("Please provide a valid search string").await?;
+        return Ok(());
+    }
+
+    if amount > SEARCHCAP || amount < 1 {
```

Comment:
You can provide min and max's to slash command arguments with `#[min = 0]` and `#[max = 20]`, if you want

See https://docs.rs/poise/latest/poise/macros/attr.command.html#input-filter-slash-only

Reply:
that's probably better than this check

Reply:
Well since it's a `prefix_command` as well, you'll probably want to keep this check. This is more of a UX thing.

Reply:
gonna remove prefix command unless we have a specific reason to keep it